### PR TITLE
Add Redis configuration

### PR DESCRIPTION
**Pull request by ameech**

### CODE REVIEW

File: tests/Configuration/RedisConfigurationTest.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+use Equip\Env;
+use PHPUnit_Framework_TestCase as TestCase;
+use Redis;
+
+class RedisConfigurationTest extends TestCase
+{
+    public function testApply()
+    {
+        $injector = new Injector;
+        $injector->delegate(Redis::class, function() {
```
Comment (shadowhand) [inline tests/Configuration/RedisConfigurationTest.php]: Nice work around for allowing `prepare` to operate normally!

**Comment by shadowhand** [comment]
:+1:

File: src/Configuration/EnvTrait.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Equip\Configuration;
+
+use Equip\Env;
+
+trait EnvTrait
```
Comment (elazar) [inline src/Configuration/EnvTrait.php]: Probably need a corresponding PR to drop [this](https://github.com/wheniwork/api-configuration/blob/master/src/EnvTrait.php) and update other files that use it in that repo.

Comment (ameech) [inline src/Configuration/EnvTrait.php]: That was the plan, figured it would work well here.

File: docs/index.md [hunk 3]
```diff
@@ -115,6 +115,7 @@ The following configurations are available but not used by default:
 
 * [`EnvConfiguration`](https://github.com/equip/framework/blob/master/src/Configuration/EnvConfiguration.php) - Use [Dotenv](https://github.com/josegonzalez/php-dotenv) to populate the content of [`Env`](https://github.com/equip/framework/blob/master/src/Env.php)
 * [`PlatesResponderConfiguration`](https://github.com/equip/framework/blob/master/src/Configuration/PlatesResponderConfiguration.php) - Use [Plates](http://platesphp.com/) as the default [responder](#responders)
+* [`RedisConfiguration`](https://github.com/equip/framework/blob/master/src/Configuration/RedisConfiguration.php) - Use Redis for in-memory store
```
Comment (ameech) [inline docs/index.md]: Good catch, updated.

File: src/Configuration/RedisConfiguration.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+{
+    use EnvTrait;
+
+    /**
+     * @inheritDoc
+     */
+    public function apply(Injector $injector)
+    {
+        $injector->prepare(Redis::class, function($redis) {
+            $redis->connect(
```
Comment (elazar) [inline src/Configuration/RedisConfiguration.php]: Is there a reason we're using the Redis extension directly as opposed to something like Predis, which would connect lazily?

Comment (shadowhand) [inline src/Configuration/RedisConfiguration.php]: HHVM builds it in. If someone wanted to use Predis they could write their own configuration.

Comment (shadowhand) [inline src/Configuration/RedisConfiguration.php]: Part of the problem is that Predis and Redis classes don't share a common interface.

Comment (elazar) [inline src/Configuration/RedisConfiguration.php]: Fair enough, just curious.

**Comment by elazar** [comment]
:+1:

**Pull request merged**